### PR TITLE
Feat/dgds 330 account panel

### PR DESCRIPTION
**Pull request by vvandijck**
This pull-request includes:
- Refactored sidebar + navigation
- Animated transitions for hiding/showing trays.
- Moved markdown styling to core styles.
- Added OIDC OAuth for login/logout.
- Added logged in account details to 'Account' panel.
- Small QoL changes.

### CODE REVIEW

File: auth/index.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
+import Oidc from 'oidc-client'
+import oidcSettings from './config'
+
+Oidc.Log.logger = console
+Oidc.Log.level = Oidc.Log.WARN
+
+const auth = new Oidc.UserManager(oidcSettings)
+
+auth.events.addUserLoaded(() => {
```
Comment (CindyvdVries) [inline auth/index.js]: Are the logs needed if not in dev?

File: components/account-details.vue [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+      <dd>{{ email }}</dd>
+      <dt>Phone number</dt>
+      <dd>{{ phone }}</dd>
+    </dl>
+    <div v-else class="account-details__placeholder">
+      <p>Please login to view your account details</p>
+    </div>
+    <div class="account-details__buttons">
+      <UiButton v-if="!user" kind="primary" @click="login">
+        Log in
```
Comment (CindyvdVries) [inline components/account-details.vue]: login, logout?
I know it is correct as verb too, but sounds weird as a button.

File: components/account-details.vue [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+import { mapState } from 'vuex'
+import auth from '../auth'
+import UiButton from './ui-button'
+
+export default {
+  components: { UiButton },
+  computed: {
+    ...mapState('preferences', ['user']),
+    name () {
+      return this.user.name || '--'
```
Comment (CindyvdVries) [inline components/account-details.vue]: Shouldn't this ( || ) or-logic be in the vuex store? also for email and phone

File: components/ui-tray.vue [hunk 4]
```diff
@@ -0,0 +1,83 @@
+<template>
```
Comment (CindyvdVries) [inline components/ui-tray.vue]: 1) panel should be updated to this style i think, but most logical to not do this in this branch.
2) need to set a scroll on the body, see about section

File: .env.example [hunk 5]
```diff
@@ -1,2 +1,10 @@
 MAPBOX_ACCESS_TOKEN=''
 SERVER_URL=''
+AUTH_URL=''
+AUTH_AUTHORITY=''
+AUTH_ID=''
+AUTH_TYPE=''
+AUTH_SCOPE=''
+AUTH_LOGIN_URL=''
```
Comment (CindyvdVries) [inline .env.example]: For later:
Remove some unnecessary vars from this .env file.

Comment (vvandijck) [reply .env.example]: Done

File: auth/index.js (hunk 1 above)
Comment (vvandijck) [reply auth/index.js]: Removed.

File: components/account-details.vue (hunk 2 above)
Comment (vvandijck) [reply components/account-details.vue]: Done

File: components/account-details.vue (hunk 3 above)
Comment (vvandijck) [reply components/account-details.vue]: No the store should just saves the `user` object, what you do with its content / how you use it is up the component. If `user` does not exist (`null`) the component shows a `--` as a placeholder. That's not up to the store to decide. If use `user` in a different component you might not want `--` to be returned.

File: components/ui-tray.vue (hunk 4 above)
Comment (vvandijck) [reply components/ui-tray.vue]: Done.

File: .env.example (hunk 5 above)
Comment (vvandijck) [reply .env.example]: Done.

**Pull request merged**